### ISSUE TITLE
DateTimePicker chops seconds and miliseconds at first use

### ISSUE THREAD

**Issue by HayoBaanAtHand** [vaadin/flow-components#1428]
I ran into a strange problem with the `DateTimePicker` component. I use a `DateTimePicker` on a form and when the form is initialized for the first time and a value is set, the seconds and millisecond are chopped off and replaced with zeros. This only happens the first time; when the form is reused, the seconds and milliseconds are maintained.

I've tried to analyze the issue by adding debug info in the value change listener and discovered that right after initially setting the value (with the correct value), the value change listener is fired again, but now with the seconds and milliseconds chopped off. When reusing the form, the value changer is only fired once (to set the new form value). When pausing execution during the second value change event, you can actually see the `DateTimePicker` showing the correct value (with seconds and milliseconds).

I've tried various things to work around the issue (setting and resetting the value in the form constructor and when reusing the form, using and not using the binder), but it still chops the seconds and milliseconds. It is as if the precision (step value) is not correctly honored at first use.

I'm using Vaadin 14.3.0 with the `DateTimePicker` configured as follows (in the form constructor):
```
    creationTimestamp = new DateTimePicker("Created at");
    creationTimestamp.setStep(Duration.ofMillis(1));
```
(the picker is actually bound to a Date field – with an appropriate conversion – but as I've also tried not using the binder this shouldn't be part of the issue)

**Comment by HayoBaanAtHand** [comment]
I finally made some time to make this into a reproducible demo application (see attached and also #1868).
[DateTimePickerBug.zip](https://github.com/vaadin/flow-components/files/6758215/DateTimePickerBug.zip)

### Steps:
- Run application from attached zip e.g. with `SPRING_PROFILES_ACTIVE=native mvn -U spring-boot:run -Dspring-boot.run.arguments='--spring.config.location=file:config/bootstrap.properties'.`
- Enter some name and password, “login”, navigate to entity page.
- Click on the third entry in the grid. The form will open with the entity. Notice that the milliseconds are chopped!
- If you now attempt to switching entries, you get a warning stating you have unsaved changes.
- Click Save (which actually doesn't really do anything but close the editor)
- Select the third grid entry again and notice the milliseconds now show as expected.

Hope this can be resolved soon as it is breaking our applications.

**Comment by sissbruecker** [comment]
Thanks for the reproduction @HayoBaanAtHand . I tested this with the newly released Vaadin 14.7.2 and can not reproduce the issue anymore. It should have been fixed as part of https://github.com/vaadin/flow-components/pull/2165

I'll close the ticket for now, please comment if you have further issues.

**Comment by HayoBaanAtHand** [comment]
14.7.2 indeed fixes this (as well as #1868).
Interestingly, I now find that the precision of the picker is only milliseconds, not micro (or nanoseconds). Since all my database date/times are in microsecond precision, I really need a time picker that is able to do microseconds (ultimately nanoseconds). Should I raise a new request for this?
